sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -156,9 +192,34 @@ public String getId() {
         if (options == null) {
             options = new CosmosContainerRequestOptions();
         }
-        return database.getDocClientWrapper()
-                   .replaceCollection(ModelBridgeInternal.getV2Collection(containerProperties), ModelBridgeInternal.toRequestOptions(options))
-                   .map(response -> ModelBridgeInternal.createCosmosAsyncContainerResponse(response, database)).single();
+
+        final CosmosContainerRequestOptions requestOptions = options;
+        return withContext(context -> replace(containerProperties, requestOptions, context)).subscriberContext(reactorContext -> {
+            Optional<String> master = reactorContext.getOrEmpty(TracerProvider.MASTER_CALL);
+            if (master.isPresent()) {
+                reactorContext = reactorContext.put(TracerProvider.NESTED_CALL, true);
+            }
+            return reactorContext.put(TracerProvider.MASTER_CALL, true);
```

Comment:
seems this code is repeated exactly, can this be factored out as a function and passed here?
\liked this
```java
return withContext(context -> replace(containerProperties, requestOptions, context))
    .subscriberContext(reactorContext -> factoredOutFunc())
```

Reply:
I was waiting on final handshake between cosmos and core for withContext api, did the refactoring now.